server/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -1529,16 +1529,21 @@ public void test_cross_join_on_top_of_fetch() throws Exception {
     @UseRandomizedSchema(random = false)
     @UseRandomizedOptimizerRules(0)
     @Test
-    public void test_ensure_hash_symbols_match_after_hash_join_is_reordered() {
+    public void test_ensure_hash_symbols_match_after_hash_join_is_reordered() throws Exception {
         execute("create table doc.t1(a int, b int)");
         execute("create table doc.t2(c int, d int)");
         execute("create table doc.t3(e int, f int)");
 
         execute("insert into doc.t1(a,b) values(1,2)");
         execute("insert into doc.t2(c,d) values (1,3),(5,6)");
         execute("insert into doc.t3(e,f) values (3,2)");
-        refresh();
+
+        execute("refresh table doc.t1");
+        execute("refresh table doc.t2");
+        execute("refresh table doc.t3");
+
         execute("analyze");
+        waitNoPendingTasksOnAll();
```

Comment:
Are you sure this is going to help? `analyze` is a blocking operation afaik, and from a quick look, I see `analyze` being executed in various tests without the wait call after it.
Maybe you can try to add iterations to validate on ci runs that it helps, before merging?

Reply:
See LookupJoinIntegrationTest

Reply:
@mkleen Could you please elaborate why this will help? Pointing to a test case were this is also used won't explain this (until I'm not missing any conversation related to that test).
Like @matriv pointed out, `analyze` should block, so I'd also quite curios why this change will help.

Reply:
Actually, I think the added `refresh` statements looks more like the real fix..